util/progress/progressui/init.go

```diff
@@ -0,0 +1,45 @@
+package progressui
+
+import (
+	"os"
+	"runtime"
+
+	"github.com/morikuni/aec"
+)
+
+var noColorIsSet bool
+var buildkitColorIsSet bool
+var colorRun aec.ANSI
+var colorCancel aec.ANSI
+var colorWarning aec.ANSI
+var colorError aec.ANSI
+var envColorString string
+
+func init() {
+	_, noColorIsSet = os.LookupEnv("NO_COLOR")
+	_, buildkitColorIsSet = os.LookupEnv("BUILDKIT_COLORS")
+
+	if noColorIsSet {
+		colorRun = aec.DefaultF
+		colorCancel = aec.DefaultF
+		colorWarning = aec.DefaultF
+		colorError = aec.DefaultF
+		return
+	} else if runtime.GOOS == "windows" {
+		colorRun = aec.CyanF
+		colorCancel = aec.YellowF
+		colorWarning = aec.Color8BitF(aec.NewRGB8Bit(255, 140, 0))
+		colorError = aec.RedF
+	} else {
+		colorRun = aec.BlueF
+		colorCancel = aec.YellowF
+		colorWarning = aec.Color8BitF(aec.NewRGB8Bit(255, 140, 0))
+		colorError = aec.RedF
```

Comment:
These colors here should probably come from shared constants in the color map?

Otherwise colors are defined in two places which might be easy to miss.